src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -63,17 +60,102 @@ import play.api.libs.json._
   * Marathon will notice spurious tasks in the offer and create the appropriate
   * unreserve operations. See https://github.com/mesosphere/marathon/issues/3223
   */
-sealed trait Task {
-  def taskId: Task.Id
-  def reservationWithVolumes: Option[Task.Reservation]
-  def runSpecVersion: Timestamp
 
-  /** apply the given operation to a task */
-  def update(update: TaskUpdateOperation): TaskUpdateEffect
+case class Task(taskId: Task.Id, runSpecVersion: Timestamp, status: Task.Status) {
+  import Task.log
 
   def runSpecId: PathId = taskId.runSpecId
 
-  def status: Task.Status
+  private[this] def hasStartedRunning: Boolean = status.startedAt.isDefined
+
+  /** apply the given operation to a task */
+  def update(instance: Instance, op: TaskUpdateOperation): TaskUpdateEffect = op match {
```

Comment:
Was this defined somewhere else before?

Reply:
Multiple `Task` subclasses had their own implementations of `update`. This one is the result of merging now deleted `updated` methods.